### ISSUE TITLE
Error installing Angular CLI globally with pnpm

### ISSUE THREAD

**Issue by aecz** [pnpm/pnpm#968]
### pnpm version:
1.23.2

### Code to reproduce the issue:
```
pnpm i -g @angular/cli@1.6.1
ng new test-pnpm
Cannot find module 'typescript'
Error: Cannot find module 'typescript'
    at Function.Module._resolveFilename (module.js:536:15)
    at Function.Module._load (module.js:466:25)
    at Module.require (module.js:579:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/aecz/.nvm/versions/node/v8.9.3/pnpm-global/1/node_modules/.registry.npmjs.org/@schematics/angular/0.1.11/@angular-devkit!schematics@0.0.42/node_modules/@schematics/angular/application/index.js:11:12)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
```

### Expected behavior:
Angular project should be created in the test-pnpm folder.

### Actual behavior:
Command fails. No folder is created.

### Additional information:

 - `node -v` prints: v8.9.3 (installed with nvm 0.33.2)
 - Windows, OS X, or Linux?: macOS High Sierra 10.13.1

**Comment by zkochan** [comment]
Maybe Angular depends on typescript? I installed typescript globally and then it worked fine

**Comment by aecz** [comment]
`@schematics/angular` does require `typescript` but its `package.json` does not list it as a dependency. It still works with npm because `@angular-devkit/build-optimizer` depends on it too and dependencies are flattened.
So not an issue with `pnpm` after all. ill create a PR for Angular CLI.

**Comment by tehlordvortex** [comment]
I seem to be experiencing something similar with `parcel`

```
pnpm install -g parcel-bundler
parcel index.html
Server running at https://localhost:1234 
🚨  Cannot find module 'posthtml-parser'
    at Function.Module._resolveFilename (module.js:489:15)
    at Function.Module._load (module.js:439:25)
    at Module.require (module.js:517:17)
    at require (/home/vortex/.nvm/versions/node/v8.3.0/pnpm-global/1/node_modules/.registry.npmjs.org/v8-compile-cache/1.1.0/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/home/vortex/.nvm/versions/node/v8.3.0/pnpm-global/1/node_modules/.registry.npmjs.org/parcel-bundler/1.4.0/node_modules/parcel-bundler/src/assets/HTMLAsset.js:2:15)
    at Module._compile (/home/vortex/.nvm/versions/node/v8.3.0/pnpm-global/1/node_modules/.registry.npmjs.org/v8-compile-cache/1.1.0/node_modules/v8-compile-cache/v8-compile-cache.js:178:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
```
running `pnpm install -g posthtml-parser` fixes that one but another error pops up about `posthtml-render`. are modules actually depending on the dependency flattening behavior of the other package managers?

the list of dependencies of `parcel-bundler` only has `posthtml`, but `post-html` itself depends on `posthtml-parser` and `posthtml-render`

**Comment by zkochan** [comment]
yeah, they do it all the time. If you see it is used in code and not declared in `package.json` then ask them to add it as a dependency. It is a bug on their end in that case.

**Comment by aecz** [comment]
Didnt find time for PR, just created the issue:
https://github.com/angular/devkit/issues/365

**Comment by aecz** [comment]
Should be fixed in the next version of `@schematics/angular` with this merged PR https://github.com/angular/devkit/pull/396. Current version is `0.1.13 `and is still failing like the above.

**Comment by aecz** [comment]
Solved with `@angular/cli` v1.6.6.
```
pnpm i -g @angular/cli@1.6.6
ng new test-pnpm
```
Works with pnpm 1.23.2 and latest 1.31.3 so it was indeed not a pnpm issue.
